### ISSUE TITLE
TextFields (and TextFormFields) should allow displaying error state without error text

### ISSUE THREAD

**Issue by rami-a** [flutter/flutter#54104]
## Use case

There are occasions where multiple `TextField` (or more likely `TextFormField`) are used on a page and the UX could call for a custom error solution instead of each textfield displaying an error message. We want to maintain the validate logic that exists on the Form but allow the developer to decide whether each textfield should show its error text.

## Proposal

Adding a new parameter to `InputDecoration` that tells TextFields they should not display their error text but still display any other error specific content (errorBorder, etc)

This would have to also work with the validate logic of `TextFormField` as well so that if a form is validated the error text can choose to not be displayed on each text field.

cc @justinmc 
Similar to https://github.com/flutter/flutter/issues/52634

**Comment by MichaelPriebe** [comment]
This is something I do too! Right now this is how I do it:
```dart
decoration: InputDecoration(
  counterStyle: const TextStyle(height: 0),
  counterText: ' ',
  errorStyle: const TextStyle(height: 0),
),
```
And then error for the validator is simply an empty string. So when there is an error all that is shown is the red error line.

**Comment by rami-a** [comment]
yup I'm using this workaround for now as well but it would be nice to have a more formal way to support this.

**Comment by roti** [comment]
The workaround from @MichaelPriebe has a problem: it causes an empty space to appear under the textfield (where the error used to be). I would be nice to have a solution without this space.

**Comment by Lancear** [comment]
Yo @rami-a!
We just copied some of the code of the TimePicker and found a Todo referencing this Issue.
Please fix it in [our project](https://github.com/scarcheek/SleepScheduler) as well if it aint already fixed and we have the same problem :)

PS: Dont take this serious lol, just some random programmers having some fun

**Comment by talski** [comment]
> The workaround from @MichaelPriebe has a problem: it causes an empty space to appear under the textfield (where the error used to be). I would be nice to have a solution without this space.

Try this: `TextStyle(height: 0.01)`, I get it working here with master.

**Comment by aytunch** [comment]
> > The workaround from @MichaelPriebe has a problem: it causes an empty space to appear under the textfield (where the error used to be). I would be nice to have a solution without this space.
> 
> Try this: `TextStyle(height: 0.01)`, I get it working here with master.

Thank you this worked by using 0.001 however this is a very ugly solution.
We need a real solution to lose error text and prevent text field outline getting shorter height.
It is crazy such a simple thing is not still solved.

**Comment by Lorenzohidalgo** [comment]
Thanks! using `TextStyle(height: 0.001)` worked to hide the error message.

It would be nice to be able to decide how to show the error message from the different options:
- not showing this at all
- showing it under the textform
- showing the error text as tooltip.

**Comment by jbryanh** [comment]
In conjunction with this, an errorLabelTextStyle property seems reasonable.
